SimG4CMS/Calo/src/HGCSD.cc

```diff
@@ -112,8 +115,8 @@ bool HGCSD::ProcessHits(G4Step * aStep, G4TouchableHistory * ) {
     // Apply fiducial cuts
     if (r/z >= slopeMin_) {
       if (getStepInfo(aStep)) {
-	if (hitExists() == false && edepositEM+edepositHAD>0.) 
-	  currentHit = createNewHit();
+	if (((hitExists() == false) || storeAllG4Hits_) && 
```

Comment:
hi @bsunanda - swap the order of the first or check to avoid calling hitExists if  storeAllG4Hits_ is true